kornia/feature/adalam/adalam.py

```diff
@@ -0,0 +1,241 @@
+from typing import Dict, Optional, Tuple
+
+import torch
+from torch import Tensor
+
+from kornia.feature.laf import get_laf_center, get_laf_orientation, get_laf_scale
+from kornia.testing import KORNIA_CHECK_SHAPE
+
+from .core import adalam_core
+from .utils import dist_matrix
+
+
+def get_adalam_default_config():
+    DEFAULT_CONFIG = {
+        'area_ratio': 100,  # Ratio between seed circle area and image area. Higher values produce more seeds with smaller neighborhoods.
+        'search_expansion': 4,  # Expansion factor of the seed circle radius for the purpose of collecting neighborhoods. Increases neighborhood radius without changing seed distribution
+        'ransac_iters': 128,  # Fixed number of inner GPU-RANSAC iterations
+        'min_inliers': 6,  # Minimum number of inliers required to accept inliers coming from a neighborhood
+        'min_confidence': 200,  # Threshold used by the confidence-based GPU-RANSAC
+        'orientation_difference_threshold': 30,  # Maximum difference in orientations for a point to be accepted in a neighborhood. Set to None to disable the use of keypoint orientations.
+        'scale_rate_threshold': 1.5,  # Maximum difference (ratio) in scales for a point to be accepted in a neighborhood. Set to None to disable the use of keypoint scales.
+        'detected_scale_rate_threshold': 5,  # Prior on maximum possible scale change detectable in image couples. Affinities with higher scale changes are regarded as outliers.
+        'refit': True,  # Whether to perform refitting at the end of the RANSACs. Generally improves accuracy at the cost of runtime.
+        'force_seed_mnn': True,  # Whether to consider only MNN for the purpose of selecting seeds. Generally improves accuracy at the cost of runtime. You can provide a MNN mask in input to skip MNN computation and still get the improvement.
+        'device': torch.device('cuda')
+        if torch.cuda.is_available()
+        else torch.device('cpu'),  # Device to be used for running AdaLAM. Use GPU if available.
+    }
+    return DEFAULT_CONFIG
+
+
+def match_adalam(
+    desc1: Tensor,
+    desc2: Tensor,
+    lafs1: Tensor,
+    lafs2: Tensor,
+    config: Dict = get_adalam_default_config(),
+    hw1: Optional[Tensor] = None,
+    hw2: Optional[Tensor] = None,
+    dm: Optional[Tensor] = None,
+) -> Tuple[Tensor, Tensor]:
+    """Function, which performs descriptor matching, followed by AdaLAM filtering (see :cite:`AdaLAM2020` for more
+    details)
+
+    If the distance matrix dm is not provided, :py:func:`torch.cdist` is used.
+
+    Args:
+        desc1: Batch of descriptors of a shape :math:`(B1, D)`.
+        desc2: Batch of descriptors of a shape :math:`(B2, D)`.
+        lafs1: LAFs of a shape :math:`(1, B1, 2, 3)`.
```

Comment:
```suggestion
        lafs1: LAFs of a shape :math:`(B1, 2, 3)`.
```
Why not like this ?

Reply:
Because our LAF convention, and all functions working with it, expect 4D tensor